### PR TITLE
feat: Enhance the support of breaking changes of openai >= 1.0

### PR DESCRIPTION
**Pull request by BeautyyuYanli**
I need some help to test:

- Official OpenAI on pro_prompt_by_openai
- Azure OpenAI on pro_prompt_by_openai
- Official OpenAI on pro_prompt_by_openai_vision

Push to the branch if any updates

### CODE REVIEW

File: tg.py [hunk 1]
```diff
@@ -40,19 +39,19 @@ def main():
...
-            openai.api_base = openai_conf.get("api_base")
-        if openai_conf.get("api_type"):
-            openai.api_type = openai_conf.get("api_type")
-        if openai_conf.get("proxy"):
-            openai.proxy = openai_conf.get("proxy")
+        openai_client = OpenAI(**openai_conf)
         print("OpenAI init done.")
-        client = OpenAI()
+    
+    azure_openai_conf: dict = config.get("azure_openai")
```
Comment (yihong0618) [inline tg.py:49]: nice design love it

**Review by yihong0618 — APPROVED** [review]

File: config.sample.toml [hunk 2]
```diff
@@ -1,23 +1,28 @@
...
 
-# Set to "azure" to use the Azure API, otherwise it will use the default OpenAI API
-# api_type = "open_ai"
+#=====================================
+# Uncomment to enable GPT rewrite by Azure OpenAI
+# [azure_opeai]
+# Find the params of openai.AzureOpenAI(...) from https://github.com/openai/openai-python
 
-# Set http proxy url like http://localhost:8080 to connect to the API through a proxy
-# proxy = ""
```
Comment (BeautyyuYanli) [inline config.sample.toml:14]: openai now use httpx to make requests, so the proxy setting is not needed to be specifictly configured now.

Comment (yihong0618) [reply config.sample.toml:14]: found it ....

Comment (yihong0618) [reply config.sample.toml:14]: and httpx win the game, https://github.com/encode amazing

**Pull request merged**